File: mediumSpringWebflux/src/main/java/com/example/mediumspringwebflux/domain/board/controller/BoardController.java
package com.example.mediumspringwebflux.domain.board.controller;

import com.example.mediumspringwebflux.domain.board.dto.BoardRequest;
import com.example.mediumspringwebflux.domain.board.dto.BoardResponse;
import com.example.mediumspringwebflux.board.service.*;
import com.example.mediumspringwebflux.domain.board.service.*;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/boards")
public class BoardController {
    private final CreateBoard createBoard;
    private final FindAllBoard findAllBoard;
    private final FindByIdBoard findByIdBoard;
    private final UpdateBoard updateBoard;
    private final DeleteBoard deleteBoard;

    public BoardController(CreateBoard createBoard, FindAllBoard findAllBoard, FindByIdBoard findByIdBoard, UpdateBoard updateBoard, DeleteBoard deleteBoard) {
        this.createBoard = createBoard;
        this.findAllBoard = findAllBoard;
        this.findByIdBoard = findByIdBoard;
        this.updateBoard = updateBoard;
        this.deleteBoard = deleteBoard;
    }

    @PostMapping
    public Mono<Void> create(@RequestBody BoardRequest request) {
        return createBoard.execute(request);
    }

    @GetMapping
    public Flux<BoardResponse> findAll() {
        return findAllBoard.execute();
    }

    @GetMapping("/{id}")
    public Mono<BoardResponse> findById(@PathVariable String id) {
        return findByIdBoard.execute(id);
    }

    @PutMapping("/{id}")
    public Mono<Void> update(@RequestBody BoardRequest request, @PathVariable String id) {
        return updateBoard.execute(id, request);
    }

    @DeleteMapping("/{id}")
    public Mono<Void> delete(@PathVariable String id) {
        return deleteBoard.execute(id);
    }

}

File: mediumSpringWebfluxcorutine/src/main/kotlin/com/example/mediumspringwebfluxcorutine/board/controller/BoardController.kt
package com.example.mediumspringwebfluxcorutine.board.controller

import com.example.mediumspringwebfluxcorutine.board.dto.BoardRequest
import com.example.mediumspringwebfluxcorutine.board.dto.BoardResponse
import com.example.mediumspringwebfluxcorutine.board.service.*
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PatchMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/boards")
class BoardController(
    private val createBoard: CreateBoard,
    private val findAllBoard: FindAllBoard,
    private val findByIdBoard: FindByIdBoard,
    private val updateBoard: UpdateBoard,
    private val deleteBoard: DeleteBoard
) {

    @GetMapping
    suspend fun findAll(): List<BoardResponse> = findAllBoard.execute()

    @GetMapping("/{id}")
    suspend fun findById(@PathVariable id: String): BoardResponse = findByIdBoard.execute(id)

    @PostMapping
    suspend fun create(@RequestBody boardRequest: BoardRequest) = createBoard.execute(boardRequest)

    @PatchMapping("/{id}")
    suspend fun update(@PathVariable id: String, @RequestBody boardRequest: BoardRequest)
    = updateBoard.execute(id, boardRequest)

    @DeleteMapping("/{id}")
    suspend fun delete(@PathVariable id: String) = deleteBoard.execute(id)

}